src/rest/handlers/rest_queries.erl

```diff
@@ -101,28 +101,100 @@ execute(Fun, Args, FieldName) ->
 
 %% @todo
 -spec reaches_naive(binary(), binary()) -> {ok, boolean()} | {error, any()}.
```

Comment:
Ten typespec się nie zgadza z tym co napisałeś w kodzie. A w ogóle, to to jest niekompatybilne z resztą modułu:

reaches_naive zostaje uruchomiona przez `execute/3`, i to `execute/3` zajmuję się liczeniem `duration`. Tak więc `reaches_naive/2` ma zwrócić `{ok, boolean()}`, a `execute/3` to opakuje w mapę `#{duration => Duration, Field => Value}`

Tak więc typespec jest ok, kod jest zły. Takie samo podejście ma być w przypadku pozostałych funkcji

Reply:
musiałem przeoczyć

Reply:
a nie, to nie mój kod